infra/build_specified_commit.py

```diff
@@ -54,42 +59,116 @@ def build_fuzzer_from_commit(project_name,
+
+  if returncode == 0:
+    #Check if commit exists
+    _, returncode = run_command_in_image(docker_image_name, [
+        '/bin/bash', '-c', 'cd ' + dir_to_check +
+        '; git fetch --unshallow; git cat-file -e ' + example_commit
+    ])
+  else:
+    _, returncode = run_command_in_image(docker_image_name, [
+        '/bin/bash', '-c',
+        'cd ' + dir_to_check + '; git cat-file -e ' + example_commit
+    ])
+  if returncode == 0:
+    return True
+  return False
+
+
+def get_repo_url(docker_image_name, dir_name):
+  """Gets a git repo URL from a specific directory in a docker image.
 
-def infer_main_repo(project_name, local_store_path, example_commit=None):
+  Args:
+    docker_image_name: The name of the image where the git repo is located
+    dir_name: The directory on the image where the git repo exists
+
+  Returns:
+    The repo URL string
+  """
+  command_to_run = 'cd /src/' + dir_name + '; git config --get remote.origin.url'
+  out, returncode = run_command_in_image(docker_image_name,
+                                         ['/bin/bash', '-c', command_to_run])
+  if not returncode and out:
+    return out.rstrip()
+  return None
+
+
+def infer_main_repo(project_name, example_commit):
   """Tries to guess the main repo a project based on the Dockerfile.
 
   NOTE: This is a fragile implementation and only works for git
   Args:
     project_name: The OSS-Fuzz project that you are checking the repo of
     example_commit: A commit that is in the main repos tree
   Returns:
-    The guessed repo url path or None on failue
+    The guessed repo url path, the repo name
   """
   if not helper.check_project_exists(project_name):
-    return None
-  docker_path = helper.get_dockerfile_path(project_name)
-  with open(docker_path, 'r') as file_path:
-    lines = file_path.read()
-    # Use generic git format and project name to guess main repo
-    if example_commit is None:
-      repo_url = re.search(
-          r'\b(?:http|https|git)://[^ ]*' + re.escape(project_name) +
-          r'(.git)?', lines)
-      if repo_url:
-        return repo_url.group(0)
-    else:
-      # Use example commit SHA to guess main repo
-      for clone_command in re.findall('.*clone.*', lines):
-        repo_url = re.search(r'\b(?:https|http|git)://[^ ]*',
-                             clone_command).group(0)
-        print(repo_url)
-        try:
-          test_repo_manager = repo_manager.RepoManager(repo_url.rstrip(),
-                                                       local_store_path)
-          if test_repo_manager.commit_exists(example_commit):
-            return repo_url
-        except:
-          pass
-    return None
+    return None, None
+  helper.build_image_impl(project_name)
+  docker_image_name = 'gcr.io/oss-fuzz/%s' % (project_name)
+
+  out, _ = run_command_in_image(docker_image_name,
+                                ['/bin/bash', '-c', 'ls /src'])
+  for dirs in out.split(' '):
```

Comment:
Should this be for `dir` in `out.split(' ')` instead of `dirs`?
How do you know they will all be seperated by spaces? `ls ~/Downloads` gives me dirs on multiple lines because i have so many files for those. Will this work?